api/v1alpha1/openapi.yaml

```diff
@@ -3009,3 +3009,157 @@ components:
       - 'TemplateDiscriminatorKubernetesSec'
       - 'TemplateDiscriminatorInlineConfig'
       - 'TemplateDiscriminatorHttpConfig'
+    CertSigningRequestApproval:
+      type: object
+      properties:
+        labels:
+          type: object
+          additionalProperties:
+            type: string
+          description: 'labels is a set of labels to apply to the device.'
+        approved:
+          type: boolean
+          description: 'approved indicates whether the request has been approved.'
+        approvedBy:
+          type: string
+          description: 'approvedBy is the name of the approver.'
+        approvedAt:
+          type: string
+          format: date-time
+          description: 'approvedAt is the time at which the request was approved.'
+      required:
+        - approved
+    CertSigningRequestCondition:
+    # this could be replaced with the pre-existing Condition schema, but Condition is not a 1:1 match to the k8s schema
+      description: 'Information about a CSR'
+      properties:
+        lastTransitionTime:
+          description: 'Time of the last status transition'
+          type: string
+          format: date-time
+        lastUpdateTime:
+          description: 'Time of the last update to current condition'
+          type: string
+          format: date-time
+        message:
+          description: 'Human readable message with details about the request state'
+          type: string
+        reason:
+          description: 'Brief reason for the request state'
+          type: string #TODO - enum?
```

Comment:
The advantages of a string are that you do not need to extend the API to add/remove options.

Reply:
True - though since the `message` field is described as human-readable (again this is modeled on k8s), that makes me think this is meant to be machine readable and should only contain expected values. But I am not sure of that.

Reply:
Looking again, maybe they are both human-readable and it's fine.

Reply:
Also I think we're actually just going to add a few variants to the `Condition` schema instead of adding this.